lib/DXIL/DxilShaderFlags.cpp

```diff
@@ -562,27 +572,23 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
             }
           }
           break;
-        case DXIL::OpCode::SampleGrad:
         case DXIL::OpCode::SampleLevel:
         case DXIL::OpCode::SampleCmpLevelZero:
-          if (!isa<Constant>(CI->getArgOperand(
-                  DXIL::OperandIndex::kTextureSampleOffset0OpIdx)) ||
-              !isa<Constant>(CI->getArgOperand(
-                  DXIL::OperandIndex::kTextureSampleOffset1OpIdx)) ||
-              !isa<Constant>(CI->getArgOperand(
-                  DXIL::OperandIndex::kTextureSampleOffset2OpIdx)))
-            hasAdvancedTextureOps = true;
+          hasAdvancedTextureOps |= hasNonConstantSampleOffsets(CI);
+          break;
+        case DXIL::OpCode::SampleGrad:
+        case DXIL::OpCode::SampleCmpGrad:
+          hasAdvancedTextureOps |= hasNonConstantSampleOffsets(CI);
+          if (!isa<UndefValue>(CI->getArgOperand(CI->getNumArgOperands() - 1)))
+            hasTiledResources = true;
           break;
         case DXIL::OpCode::Sample:
         case DXIL::OpCode::SampleBias:
         case DXIL::OpCode::SampleCmp:
-          if (!isa<Constant>(CI->getArgOperand(
-                  DXIL::OperandIndex::kTextureSampleOffset0OpIdx)) ||
-              !isa<Constant>(CI->getArgOperand(
-                  DXIL::OperandIndex::kTextureSampleOffset1OpIdx)) ||
-              !isa<Constant>(CI->getArgOperand(
-                  DXIL::OperandIndex::kTextureSampleOffset2OpIdx)))
-            hasAdvancedTextureOps = true;
+        case DXIL::OpCode::SampleCmpBias:
```

Comment:
Is there not an additional feature flag for using `SampleCmpBias` (and `SampleCmpGrad`) in the first place, or is it a required part of SM 6.8?

Reply:
These 2 are required by SM6.8.